kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/MetricsReportSuite.scala

```diff
@@ -0,0 +1,379 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.defaults
+
+import java.io.File
+import java.util
+import java.util.{Objects, Optional}
+
+import scala.collection.mutable.ArrayBuffer
+
+import io.delta.golden.GoldenTableUtils.goldenTablePath
+import io.delta.kernel.defaults.utils.TestUtils
+import io.delta.kernel.engine.{CommitCoordinatorClientHandler, Engine, ExpressionHandler, FileSystemClient, JsonHandler, MetricsReporter, ParquetHandler}
+import io.delta.kernel.metrics.{MetricsReport, SnapshotReport}
+import io.delta.kernel.{Snapshot, Table}
+import io.delta.kernel.internal.fs.Path
+import io.delta.kernel.internal.metrics.Timer
+import io.delta.kernel.internal.util.FileNames
+import org.scalatest.funsuite.AnyFunSuite
+
+/**
+ * Test suite to test the Kernel-API created [[MetricsReport]]s. This suite is in the defaults
+ * package to be able to use real tables and avoid having to mock both file listings AND file
+ * contents.
+ */
+class MetricsReportSuite extends AnyFunSuite with TestUtils {
+
+  ///////////////////////////
+  // SnapshotReport tests //
```

Comment:
How many different types of reports do you anticipate we will have? If test + helper code for just SnapshotReport is ~300 LOC, I wonder if we should split this up into different files?

e.g. `SnapshotMetricsReportSuite extends MetricsReportTestUtils`